controllers/comment_controller.go

```diff
@@ -3,111 +3,318 @@ package controllers
 import (
 	"context"
 	"fmt"
-	"jitD/models"
+	configs "jitD/configs"
+	models "jitD/models"
+	"log"
 	"net/http"
+	"time"
+
+	"cloud.google.com/go/firestore"
 
 	"github.com/gin-gonic/gin"
+	mapstructure "github.com/mitchellh/mapstructure"
+)
 
-	configs "jitD/configs"
-	"log"
+// service Create Comment
+func CreateComment(c *gin.Context) {
+	ctx := context.Background()
+	client := configs.CreateClient(ctx)
+	comment := models.Comment{}
 
-	mapstructure "github.com/mitchellh/mapstructure"
+	if err := c.BindJSON(&comment); err != nil {
+		log.Fatalln(err)
+		return
 
-	"google.golang.org/api/iterator"
-)
+	}
+	fmt.Println("time.Now().UTC(): ", time.Now().Format(time.RFC3339))
+
+	// assigning comment object to comment model
+	comment.Content = string(comment.Content)
+	comment.Like = []*firestore.DocumentRef{}
+	comment.UserId = string(comment.UserId)
+	comment.Date.Format(time.RFC3339)
+	//currentTime := time.Now().Format(time.RFC3339)
+
+	//add comment to comment collection
+	commentRef, _, err := client.Collection("Comment").Add(ctx, comment)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Cant to create comment",
+		})
+	} else {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "create comment success",
+		})
+	}
+
+	//------- Updateing to post --------------
+
+	post_id := c.Param("post_id")
+	post := models.Post{}
+	dsnap, err2 := client.Collection("Post").Doc(post_id).Get(ctx)
+	if err2 != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Cant to find postid",
+		})
+	}
+	mapstructure.Decode(dsnap.Data(), &post)
+	post.Comment = append(post.Comment, commentRef)
+	setData, _ := client.Collection("Post").Doc(post_id).Set(ctx, post)
+	if err2 != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Cant to find postid",
+		})
+	}
+	c.JSON(http.StatusOK, setData)
```

Comment:
I think this line should to remove, `c.JSON(http.StatusOK, setData)`  it should append the end of function (line 88)

[ in other this, that look to good ]